### DISCUSSION TITLE
Rethrow in finally not allowed

### DISCUSSION THREAD

**Discussion by AnakinRaW** [dotnet/csharplang#4660] [Q&A]
Hey, while reading through the ECMA-335 i noticed that `finally` clauses are allowed to rethrow when within a try-catch block. 

However when trying to create such a code in C# i get an compiler error CS0724 which explicitly forbids this situation. I'd like to know what was the reason not enabling this in C# 

Cheers

```cs
public void M() { 
        try{
        } catch {
          try{
          } finally{
              throw; // CS0724 here
          }   
        }        
    }
```

**Comment by canton7** [comment]
What if the `try` block doesn't throw an exception? What exception should that `finally` block throw?

**Reply by HaloFour** [reply]
I think the intent is to throw the exception caught by the `catch` clause of the outer `try` block.  I'd suggest it's not allowed because it would be obnoxious to change the syntax to be context dependent in that case and that the workaround using flags is sufficient given how uncommon it would be to want this.

**Reply by canton7** [reply]
Ah, yes you're right.

Note that `ExceptionDispatchInfo` exists these days, which gives you a way to re-throw any exception from any place without losing the stack trace, without needing to use `throw;` from within a `catch` block specifically.
